cl/recap/mergers.py

```diff
@@ -622,14 +692,7 @@ def add_docket_entries(d, docket_entries, tags=None):
 
         de.description = docket_entry["description"] or de.description
         date_filed = docket_entry["date_filed"]
-        if isinstance(date_filed, datetime):
-            # For now we do dumb date conversion. This simply returns a date
-            # object with the same year, month, and day, ignoring time and
-            # timezones. Once the DB is upgraded to support timezones, we can
-            # do better.
-            date_filed = date_filed.date()
-
-        de.date_filed = date_filed or de.date_filed
+        set_docket_entry_date_and_time(de, date_filed)
```

Comment:
I think a better API here would be to have a function that took the court ID and a `datetime | date` and then gave you back the corrected date/datetime. The way you have it now, `set_docket_entry_date_and_time` does both that work and the work of setting the value. I thin it'd be much more clear here to have a line that says:

```
de.date_filed, de.time_filed = localize_date_and_time(court_id, date_filed)
```

That's a lot more clear, isn't it, and I think it'd give you better tests too?

Reply:
Thanks, this was really helpful, now the method is called `localize_date_and_time` and only receives the court and the `date_filed` and returns the localized `date_filed` and `time_filed` (or None if the instance is not `datetime`)

So now using this method in other places where it fits very well.